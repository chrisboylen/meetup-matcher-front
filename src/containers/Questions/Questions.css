.user-questions {
  background-color: #bcb1b1d6;
  border: 0;
  border-radius: 12px;
  display: flex;
  flex-direction: column;
  padding: 0;
  margin: 2rem auto;
  width: 30rem; }
  .user-questions h2 {
    background-color: #fef5f5d9;
    border-radius: 12px 12px 0 0;
    color: #ff0000;
    text-align: center;
    margin: 0;
    padding: .8rem; }
  .user-questions .question-btn {
    border-radius: 10px;
    display: block;
    margin: 2rem auto 1rem; }

label {
  display: block;
  padding-left: 2rem; }
  label input {
    box-shadow: none;
    height: 1.5rem;
    padding: 0;
    margin: 0 1rem 0 0;
    vertical-align: bottom;
    position: relative;
    top: .7rem;
    width: 1.5rem; }

input[type="checkbox"]:checked + label {
  background-color: #f50a0a;
  color: red;
  outline: 0; }

label:hover {
  cursor: pointer; }
