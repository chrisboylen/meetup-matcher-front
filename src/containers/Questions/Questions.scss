$bkgWhite: #fef5f5d9;
$mainRed: #ff0000;

.user-questions {
  background-color: #bcb1b1d6;
  border: 0;
  border-radius: 12px;
  display: flex;
  flex-direction: column;
  padding: 0;
  margin: 4rem auto;
  width: 30rem;
    h2 {
      background-color: $bkgWhite;
      border-radius: 12px 12px 0 0;
      color: $mainRed;
      text-align: center;
      margin: 0 0 1rem 0;
      padding: .8rem;
    }
    .question-btn {
      border-radius: 10px;
      display: block;
      margin: 2rem auto 1rem;
    }
}

label {
  display: block;
  padding-left: 2rem;
    input {
      width: 1.5rem;
      height: 1.5rem;
      padding: 0;
      margin: 0 1rem 0 0;
      vertical-align: bottom;
      position: relative;
      top: .7rem;
      *overflow: hidden;
    }
}

input[type="checkbox"]:checked +label {
  background-color: #f50a0a;
  color: red;
  outline: 0;
}

label:hover {
  cursor: pointer;
}
