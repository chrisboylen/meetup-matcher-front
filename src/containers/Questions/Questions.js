import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { addUserQuestions } from '../../actions';
import './Questions.css';

export class Questions extends Component {
  constructor () {
    super();
    this.state = {
      traveling: false,
      entrepreneur: false,
      softwareDevs: false,
      hiking: false,
      photography: false,
      diningout: false,
      realEstate: false
    }
  }

  handleSubmit = (e) => {
    e.preventDefault();
    
    const { addUserQuestions } = this.props;
    const { questions } = this.state;
console.log(this.props.questions);
    addUserQuestions(questions);
  }

  handleChange = (e) => {
    const { name } = e.target;

    this.setState({ [name]: !this.state[name] });
  }

  render() {
    const {
      traveling,
      entrepreneur,
      softwareDevs,
      hiking,
      photography,
      diningout,
      realEstate
    } = this.state;

    return (
      <form 
        className="user-questions"
        onSubmit={this.handleSubmit}
      >
        <h2>Select your favorite meetup topics</h2>
        <label htmlFor="traveling">
          <input 
            type="checkbox" 
            name="traveling" 
            id="traveling"
            checked={traveling}
            value={traveling}
            onChange={this.handleChange}
          /> 
          Traveling
        </label>
        <label htmlFor="entrepreneur">
          <input 
            type="checkbox" 
            name="entrepreneur" 
            id="entrepreneur"
            checked={entrepreneur}
            value={entrepreneur}
            onChange={this.handleChange}
          /> 
          Entrepreneur
        </label>
        <label htmlFor="softwareDevs">
          <input 
            type="checkbox" 
            name="softwareDevs" 
            id="softwareDevs"
            checked={softwareDevs}
            value={softwareDevs}
            onChange={this.handleChange}
          /> 
          Software Developers
        </label>
        <label htmlFor="hiking">
          <input 
            type="checkbox" 
            name="hiking" 
            id="hiking"
            checked={hiking}
            value={hiking}
            onChange={this.handleChange}
          /> 
          Hiking
        </label>
        <label htmlFor="photography">
          <input 
            type="checkbox" 
            name="photography" 
            id="photography"
            checked={photography}
            value={photography}
            onChange={this.handleChange}
          /> 
          Photography
        </label>
        <label htmlFor="diningout">
          <input 
            type="checkbox" 
            name="diningout" 
            id="diningout"
            checked={diningout}
            value={diningout}
            onChange={this.handleChange}
          /> 
          Dining Out
        </label>
        <label htmlFor="realEstate">
          <input 
            type="checkbox" 
            name="realEstate" 
            id="realEstate"
            checked={realEstate}
            value={realEstate}
            onChange={this.handleChange}
          /> 
          Real Estate
        </label>
        <button className="question-btn">bingo</button>
      </form>
    )
  }
}

Questions.propTypes = {
  addUserQuestions: PropTypes.func,
};

export const mapStateToProps = ({ questions }) => ({ questions });

export const mapDispatchToProps = (dispatch) => ({
  addUserQuestions: (questions) => dispatch(addUserQuestions(questions))
});

export default connect(mapStateToProps, mapDispatchToProps)(Questions);