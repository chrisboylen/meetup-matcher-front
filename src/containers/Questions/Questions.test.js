import React from 'react';
import { shallow } from 'enzyme';
import { Questions, mapStateToProps, mapDispatchToProps } from './Questions';

describe('Questions', () => {
  let wrapper;
  let mockEvent;

  beforeEach(() => {
    wrapper = shallow(<Questions />);
    mockEvent = {target: {value: 'yes'}};
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should set state when invoked', () => {
      mockEvent = {target: {name: 'traveling'}};
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().traveling).toBe(true);

      mockEvent = {target: {name: 'entrepreneur'}};
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().entrepreneur).toBe(true);

      mockEvent = {target: {name: 'softwareDevs'}};
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().softwareDevs).toBe(true);

      mockEvent = {target: {name: 'hiking'}};
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().hiking).toBe(true);

      mockEvent = {target: {name: 'photography'}};
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().photography).toBe(true);

      mockEvent = {target: {name: 'diningout'}};
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().diningout).toBe(true);

      mockEvent = {target: {name: 'realEstate'}};
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().realEstate).toBe(true);
    });
    it('should invoke handleChange when traveling is changed', () => {
      const spy = spyOn(wrapper.instance(), 'handleChange');
      wrapper.instance().forceUpdate();
      wrapper.find('#traveling').simulate('change', mockEvent);

      expect(spy).toHaveBeenCalled();
    });
    it('should invoke handleChange when entrepreneur is changed', () => {
      const spy = spyOn(wrapper.instance(), 'handleChange');
      wrapper.instance().forceUpdate();
      wrapper.find('#entrepreneur').simulate('change', mockEvent);

      expect(spy).toHaveBeenCalled();
    });
  });
});