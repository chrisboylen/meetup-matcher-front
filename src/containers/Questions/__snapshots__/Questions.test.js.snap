// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions should match snapshot 1`] = `
<div
  className="question-cont"
>
  <withRouter(Connect(Header)) />
  <form
    className="user-questions"
    onSubmit={[Function]}
  >
    <h2>
      Select your favorite meetup topics
    </h2>
    <label
      htmlFor="traveling"
    >
      <input
        checked={false}
        id="traveling"
        name="traveling"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
      Traveling
    </label>
    <label
      htmlFor="entrepreneur"
    >
      <input
        checked={false}
        id="entrepreneur"
        name="entrepreneur"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
      Entrepreneur
    </label>
    <label
      htmlFor="softwareDevs"
    >
      <input
        checked={false}
        id="softwareDevs"
        name="softwareDevs"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
      Software Developers
    </label>
    <label
      htmlFor="hiking"
    >
      <input
        checked={false}
        id="hiking"
        name="hiking"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
      Hiking
    </label>
    <label
      htmlFor="photography"
    >
      <input
        checked={false}
        id="photography"
        name="photography"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
      Photography
    </label>
    <label
      htmlFor="diningout"
    >
      <input
        checked={false}
        id="diningout"
        name="diningout"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
      Dining Out
    </label>
    <label
      htmlFor="realEstate"
    >
      <input
        checked={false}
        id="realEstate"
        name="realEstate"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
      Real Estate
    </label>
    <button
      className="question-btn"
    >
      SAVE
    </button>
  </form>
</div>
`;
